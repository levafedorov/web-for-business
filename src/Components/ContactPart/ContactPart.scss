$white: white;
$black: black;
$dark-gray: #898d8e;
$light-gary: #b7bcbe;
$orange: #d7794a;
$whiten-gray:#e8e6e6;
$swampy: #1f3138;
$red: rgba(179, 38, 38, 0.705);

@mixin decorator($bg:none, $bord:none){
   background-color: $bg;
   border: $bord;
}


@mixin bg-image($bg){
  background-image: url($bg);
  background-repeat: no-repeat;
  background-size: cover;
}



@mixin flex($just:center, $al:center){
   display: flex;
     justify-content: $just; 
     align-items: $al;
  }

  @mixin flex-self($just:center, $al:center){
     justify-self: $just; 
     align-self: $al;
  }


  @mixin myriad($size:1em, $height:1em, $weight:normal){
    font-family: 'Myriad Pro Regular', 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
    font-size: $size;
    line-height: $height;
    font-weight: $weight;
}

@mixin baskerville($size:1em, $height:1em, $weight:normal){
   font-family: 'Baskerville Regular', 'Times New Roman', Times, serif;
   font-size: $size;
   line-height: $height;
   font-weight: $weight;
}

@mixin athelas($size:1em, $height:1em, $weight:normal){
   font-family: 'Athelas Regular', Georgia, 'Times New Roman', Times, serif;
   font-size: $size;
   line-height: $height;
   font-weight: $weight;
}

@mixin quatro($size:1em, $height:1em, $weight:normal){
   font-family: 'Quatro Regular' 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
   font-size: $size;
   line-height: $height;
   font-weight: $weight;
}

@mixin helvetica($size:1em, $height:1em, $weight:normal){
   font-family: Helvetica, Arial, sans-serif;
   font-size: $size;
   line-height: $height;
   font-weight: $weight;
}

@mixin padd-def($bot: 20px, $top: 20px, $left:0px, $right:0px){
   padding-top: $top;
   padding-bottom: $bot;
   padding-left: $left;
   padding-right: $right;
}




.contact-page{
    
    @include flex();
    .bg{
        @include flex(); 
        width: 100%;
        min-height: 1438px;
    }
    .contact-place{
        @include flex();
        flex-direction: column;
        flex-basis: 1148px;
        height: 1100px;
        padding:0 10px;
    
    }
   .contact-text-intro{
       padding-top: 1px;
       max-width: 848px;
       height: 356px;
       h2, p {
           text-align: center;
       }
       h2{ @include athelas(2.3em, 2.3em);
           text-transform: uppercase;}
       p{ @include myriad(1em, 1.6em); 
          margin-top: 60px;} 
   }
   .contact-form{
       width: 100%;
       box-sizing: border-box;
       @include padd-def();
       margin:0;
       padding:0;
       & > * {
           width: 100%;
       }

        input, textarea, select{
            box-sizing: border-box;
            background-color:transparent;
            @include myriad(22px, 25px);
            padding-left: 20px;
            padding-right: 20px;
            outline: none;
            border:none;
            border-bottom: 1px solid black;
            resize: none;
            
        }
       @include flex(flex-start, flex-start);
        flex-direction: column;
       .sides{  
        padding-bottom: 10px;
        @include flex(center, stretch);
        flex-wrap: wrap;

        .left-side, .right-side{
           @include flex(space-between, stretch);
           flex-direction: column;
           min-width:0px;
           margin: 11px;
           & > *{
               margin-top: 20px;
           }
            input, select { 
                box-sizing: border-box;
                width: 100%;
                height: 55px;
            }
        }
         .left-side{
             width: 48%;
           .name-container{
               #name{

               }
           }
           .phone-container{
            #phone-number{
                
            }
        }
         }
         .right-side{
            width: 48%;
              .category-container{
                  #category{

                  }
              }
              .city-container{
                #city{
                    
                }
            }
        }
       }
       .mail-container{
        margin: 5px;
        width: 99%;
          #mail{ 
            box-sizing: border-box;
            width: 100%;
            height: 55px;
        }
       }
       .message-container{
        margin: 5px;
        width: 99%; 
        margin-top: 20px;
        height:193px;
        #message{
              height: 100%;
              width: 100%;
        }
       }
    .container.error input, .container.error select, .container.error textarea{
        border-bottom: 1px solid $red;
        transition: none;
    }

    .container.error{
        color: $red;
        position: relative;
        transition: none;
      .error-in-form{
          position: absolute;
          font-size: 14px;
          line-height: 10px;
          top: -17px;
          transition: none;
      }
    }

    .mail-container.error .error-in-form{
        top:-14px;
    }

   }
   .contact-part-btn{
       align-self: flex-end;
       margin-top: 100px;
       @include myriad(0.8em, 0.8em);
       border-radius: 2px;
   }
}









.contact-page.default{
     @include bg-image('../../images/backgrounds/forest-bg1.jpg');
     .bg{background-color: transparentize($swampy, 0.1);}
     .contact-text-intro h2{
         color:$white;
     }
     input, textarea, select, p{
         color: $light-gary;
         &::placeholder{
            color: $light-gary;
        }
     }
     select option{
         color: $black;
     }

     .contact-part-btn{background-color: $orange;
                       color: $white;
                      border:2px  solid darken($orange, 20%)}
}


.contact-page.dark{
    background-color: $black;
    .contact-text-intro h2{
        color:$orange;
    }
    input, textarea, select, p{
        color: $white;
        &::placeholder{
           color: $white;
       }
    }
    select option{
        color: $black;
    }

    .contact-part-btn{background-color: $orange;
                      color: $white;
                     border:2px solid $white;}
    .contact-form {
        input, textarea, select{
            border-bottom:1px solid $orange;
        }  
    }                
}


.contact-page.light{
    @include bg-image('../../images/backgrounds/forest-bg1.jpg');
     .bg{background-color: transparentize($swampy, 0.1);}
    .contact-text-intro h2{
        color:$orange;
    }
    input, textarea, select, p{
        color: $white;
        &::placeholder{
           color: $white;
       }
    }
    select option{
        color: $black;
    }

    .contact-part-btn{background-color: $orange;
                      color: $white;
                     border:2px solid $white;}
    .contact-form {
        input, textarea, select{
            border-bottom:1px solid $orange;
        }  
    }                
}








@media screen and (max-width: 900px){
    .contact-page .contact-form .sides .left-side, .contact-page .contact-form .sides .right-side{
        width: 80%;
    }

}






 @media screen and (max-width: 500px) and (max-height: 700px){
    .contact-page{
    .bg{min-height: 1000px;}
    .contact-text-intro{
        height: 300px;
        p{ margin-top: 0px;}
    }
    .contact-form .sides {
        .right-side, .left-side{
            margin:0;
        }
        .left-side input,  .left-side select, 
        .right-side input,  .right-side select{
        height: 25px;
        }
    }
    .contact-form .mail-container #mail{
        height: 25px;
    }

    .contact-part-btn{
        margin-top: 20px;
        height: 50px;
        width: 150px;}

    }
   
    }
    
  
 