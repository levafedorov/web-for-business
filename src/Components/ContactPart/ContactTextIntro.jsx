import React from 'react';



export default function ContactTextIntro(props){
    

    return <div className="contact-text-intro">
        <h2>Contact Us</h2>
        <p>Lorem, ipsum dolor sit amet consectetur adipisicing elit. Maiores mollitia
             vel sapiente natus
              assumenda soluta earum explicabo, excepturi fugit minus.
        </p>
     </div>    
}