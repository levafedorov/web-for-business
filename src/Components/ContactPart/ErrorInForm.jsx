import React from 'react';
import Button from '../Buttons/Button'
import { useEffect, useState } from 'react';


export default function ErrorInForm(props){
    
  


    return <p className="error-in-form">
           {props.message}
          </p>
 
}