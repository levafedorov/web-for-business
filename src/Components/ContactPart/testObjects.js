export const testState = {
    form: {name: '',
        phone: '+420',
        category:'First',
        city:'Praha',
        mail:'',
        message:'',},
    errObj:{
        name: '',
        phone: '',
        category:'',
        city:'',
        mail:'',
        message:''
    }
      }

export  const testFormFilled = {
    form: {name: 'Hello There',
        phone: '+420775645657',
        category:'First',
        city:'Praha',
        mail:'qwery@qwery',
        message:'Message',},
    errObj:{
        name: '',
        phone: '',
        category:'',
        city:'',
        mail:'',
        message:''
    }
        }

export  const testFormErrors = {
    form: {name: 'Hello There',
        phone: '+420775645657',
        category:'First',
        city:'Praha',
        mail:'qwery@qwery',
        message:'Message',},
    errObj:{
        name: 'Error here',
        phone: 'Error here',
        category:'Error here',
        city:'Error here',
        mail:'Error here',
        message:'Error here'
    } }

export const defaultState = {
    form: {name: '',
        phone: '+420',
        category:'First',
        city:'Praha',
        mail:'',
        message:'',},
    errObj:{
        name: '',
        phone: '',
        category:'',
        city:'',
        mail:'',
        message:''
    }
        }
    
export const validObjects = [{
    name: 'Gven Bor≈°',
    phone: '+420445665345',
    category:'First',
    city:'Praha',
    mail:'qwery@gmail.com',
    message:'+71912205321',},
    {name: ' Fedorov  Lev',
    phone: '+783456723322',
    category:'second',
    city:'Brno',
    mail:'fls---@mail.ru',
    message:'Hello there',},
    {name: 'Bilbo Baggins',
    phone: '+3112231234454',
    category:'Third',
    city:'Plzen',
    mail:'asds_asdy@yandex.ru',
    message:'Hello there',
}];

export const invalidObjects = [{
    name: 'Kivi123',
    phone: '+420445665345',
    category:'First',
    city:'Praha',
    mail:'qwery@gmail.com',
    message:'+71912205321',},
    {name: 'Fedorov Lev',
    phone: '+7834534',
    category:'second',
    city:'Brno',
    mail:'mail.ru',
    message:'',},
    {name: 'Nota fa--*',
    phone: '+3112231234454',
    category:'Third',
    city:'Plzen',
    mail:'asdsasdy@yandex.ru',
    message:'', }];
            

export const tryStateObj = { 
    name: 'Name',
    phone: '+420745234',
    category:'Second',
    city:'Brno',
    mail:'qwert@qwery',
    message:'Message here hello'  }