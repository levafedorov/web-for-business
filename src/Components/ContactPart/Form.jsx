import React, {Component} from 'react';
import ErrorInForm from './ErrorInForm';
import PropTypes from "prop-types";




export default class Form extends Component{ 
     

   

   textHandler = (ev) => {
       const target = ev.target;
       this.props.setFormState({[target.name]: target.value}, 'form');
      
   }

    render(){
        const {name, message, phone, city, category, mail, errObj} = this.props;
        
      

return <form action="" method="" className="contact-form">
          <div className="sides">
                <div className="left-side">

                    <div className={`name-container container ${errObj.name ? 'error' : null}`}>
                        {errObj.name ? <ErrorInForm message={errObj.name}/> : null}
                        <input type="text" id="name" name="name" value={name} onChange={this.textHandler} placeholder="Name" className="form-input"/>
                    </div>  

                    <div className={`phone-container container ${errObj.phone ? 'error' : null}`}>
                        {errObj.phone ? <ErrorInForm message={errObj.phone}/> : null}
                        <input type="tel" id="phone-number" name="phone" value={phone} onChange={this.textHandler} className="form-input"/>
                    </div>  

                </div>

                    <div className="right-side">
                        
                            <div className="category-container container">
                                <select name="category" id="category" value={category} onChange={this.textHandler} className="form-input">
                        {this.props.categories.map(item => <option value={item} key={item}>{item}</option>)}
                                </select>
                            </div>    

                            <div className="city-container container">
                                <select name="city" id="city" value={city} onChange={this.textHandler}>
                        {this.props.cities.map(item => <option value={item} key={item}>{item}</option>)}
                                </select>
                            </div> 

                  </div>

          </div>

            <div className={`mail-container container ${errObj.mail? 'error' : null}`}>
            {errObj.mail ? <ErrorInForm message={errObj.mail}/> : null}
            <input type="mail" id="mail" name="mail" placeholder="mail@mail" value={mail} onChange={this.textHandler} className="form-input"/>
            </div>

            <div className="message-container container">
                <textarea name="message" id="message" cols="30" rows="10" placeholder="...message" value={message} onChange={this.textHandler} className="form-input">
                  
               </textarea>
            </div>
    </form>
    }
}

Form.propTypes = {
    form: PropTypes.objectOf(PropTypes.string),
    errObj: PropTypes.objectOf(PropTypes.string)
}

Form.defaultProps = {
    categories: ["First", "Second", "Third"],
    cities:["Praha", "Brno", "Plzen"]
}