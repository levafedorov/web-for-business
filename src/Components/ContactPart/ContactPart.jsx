import React, {Component} from 'react';
import './ContactPart.scss';
import ContactTextIntro from './ContactTextIntro';
import PageThemes from '../PageContexts/PageThemes';
import Form from './Form';
import Button from '../Buttons/Button';
import XRegExp from 'xregexp';


export default class ContactPart extends Component{

    state = {
      form: {name: '',
        phone: '+420',
        category:'First',
        city:'Praha',
        mail:'',
        message:'',},
      errObj:{
        name: '',
        phone: '',
        category:'',
        city:'',
        mail:'',
        message:''
      }
    }
   
   setFormState = (obj, name) => {
    this.setState({[name]: Object.assign(this.state[name], obj)});
   }

   

  sendForm = (ev) => {
   const  {name, phone, category, city, mail, message} = this.state.form;

   const unicodeWord = XRegExp('^\\pL+$', 'x');
   const phoneReg = new RegExp('^[+][0-9]{10,}');
   const mailReg = new RegExp('^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6}$');

   let toSend = true;
   let errObj = {};

   function validate(key, elem, regexp, message){
    if(!elem){
        errObj[key] = `PLease fill ${key} field`;
        toSend = false;
    }else if(regexp && !regexp.test(elem.split(' ').join(''))){ 
        errObj[key] = message;
        toSend = false;
    }else{ errObj[key] = '';}  
   }

   validate('name', name, unicodeWord, 'the field has to contain only letters');
   validate('phone', phone, phoneReg, 'the field has to contain more then 10 digits and code of the state');
   validate('mail', mail, mailReg, 'incorrect form of the eMail');
   validate('category', category);
   validate('city', city);

   if(toSend){
       alert(JSON.stringify(this.state.form));
   }else{
         this.setFormState(errObj, 'errObj');
   }
}



static contextType = PageThemes;

    render(){
      let className = "contact-page stretched transition-all "
      className += this.context.theme;

        const categories = ['Fisrt', 'Second', 'Third'];
        const cities = ['Praha', 'Brno', 'Plzen', 'Ostrava', 'Olomouc'];
        return <div className={className} id="contacts">
          <div className="bg">
          <div className="contact-place">
              <ContactTextIntro/>
              <Form categories={categories} cities={cities} setFormState={this.setFormState} errObj={this.state.errObj} {...this.state.form} />
              <Button class={'contact-part-btn'} fn={this.sendForm}>Contact us</Button>
              </div>
          </div>
        </div>
    }
}
