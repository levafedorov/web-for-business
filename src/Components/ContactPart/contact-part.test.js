import React from 'react';
import {mount, shallow} from 'enzyme';
import ContactPart from "./ContactPart";
import ThemeContext from "../PageContexts/PageThemes";
import {defaultState, validObjects, invalidObjects, tryStateObj} from "./testObjects";






describe("ContactPart component", () => {
   let component = shallow(<ContactPart/>);
   let instance = component.instance();
   let mockedSetFormState;

   beforeAll(()=>{
    global.alert = jest.fn();}
   );
 
   afterEach(() => {
    global.alert.mockRestore();
    component.setState(defaultState);
   });

   describe("Check context", () => {
       
      test("Check with context", () => {
        const component = mount(
            <ThemeContext.Provider value={{theme: 'light', change: () => {}}}><ContactPart/></ThemeContext.Provider>)
        expect(component).toMatchSnapshot();
      });

      test("Check without context", () => {
        const component = mount(<ContactPart/>);
        expect(component).toMatchSnapshot();
       });
      });

    describe("Check methods", () => {
        describe("Check setFormState", () => {
             
            test("Check if setFormState fills form states", () => {
                 expect(component.state()).toMatchObject(defaultState);
                 instance.setFormState(tryStateObj, "form");
                 expect(component.state().form).toMatchObject(tryStateObj);
            });

            test("Check if setFormState fills form error object", () => {
                expect(component.state()).toMatchObject(defaultState);
                instance.setFormState(tryStateObj, "errObj");
                expect(component.state().errObj).toMatchObject(tryStateObj);
            });

        });
        describe("Check sendForm", () => {
   
            test("positive tests", () => {
                for(let int = 0; int < validObjects.length; int++){
                    component.setState({form: validObjects[int]});
                    instance.sendForm();
                    expect(global.alert).toHaveBeenCalledTimes(1);
                    global.alert.mockRestore();
                }
            });
             test("negative tests", () => {
                for(let int = 0; int < invalidObjects.length; int++){
                    let component = mount(<ContactPart/>);
                    component.instance().setFormState = jest.fn();
                    component.setState({form: invalidObjects[int]});
                    component.instance().sendForm();
                    expect(component.instance().setFormState).toHaveBeenCalledTimes(1);
    
                }
             });
        });

    });

   });



