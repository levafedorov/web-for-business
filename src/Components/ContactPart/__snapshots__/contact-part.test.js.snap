// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPart component Check context Check with context 1`] = `
<ContactPart>
  <div
    className="contact-page stretched transition-all light"
    id="contacts"
  >
    <div
      className="bg"
    >
      <div
        className="contact-place"
      >
        <ContactTextIntro>
          <div
            className="contact-text-intro"
          >
            <h2>
              Contact Us
            </h2>
            <p>
              Lorem, ipsum dolor sit amet consectetur adipisicing elit. Maiores mollitia vel sapiente natus assumenda soluta earum explicabo, excepturi fugit minus.
            </p>
          </div>
        </ContactTextIntro>
        <Form
          categories={
            Array [
              "Fisrt",
              "Second",
              "Third",
            ]
          }
          category="First"
          cities={
            Array [
              "Praha",
              "Brno",
              "Plzen",
              "Ostrava",
              "Olomouc",
            ]
          }
          city="Praha"
          errObj={
            Object {
              "category": "",
              "city": "",
              "mail": "",
              "message": "",
              "name": "",
              "phone": "",
            }
          }
          mail=""
          message=""
          name=""
          phone="+420"
          setFormState={[Function]}
        >
          <form
            action=""
            className="contact-form"
            method=""
          >
            <div
              className="sides"
            >
              <div
                className="left-side"
              >
                <div
                  className="name-container container null"
                >
                  <input
                    className="form-input"
                    id="name"
                    name="name"
                    onChange={[Function]}
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="phone-container container null"
                >
                  <input
                    className="form-input"
                    id="phone-number"
                    name="phone"
                    onChange={[Function]}
                    type="tel"
                    value="+420"
                  />
                </div>
              </div>
              <div
                className="right-side"
              >
                <div
                  className="category-container container"
                >
                  <select
                    className="form-input"
                    id="category"
                    name="category"
                    onChange={[Function]}
                    value="First"
                  >
                    <option
                      key="Fisrt"
                      value="Fisrt"
                    >
                      Fisrt
                    </option>
                    <option
                      key="Second"
                      value="Second"
                    >
                      Second
                    </option>
                    <option
                      key="Third"
                      value="Third"
                    >
                      Third
                    </option>
                  </select>
                </div>
                <div
                  className="city-container container"
                >
                  <select
                    id="city"
                    name="city"
                    onChange={[Function]}
                    value="Praha"
                  >
                    <option
                      key="Praha"
                      value="Praha"
                    >
                      Praha
                    </option>
                    <option
                      key="Brno"
                      value="Brno"
                    >
                      Brno
                    </option>
                    <option
                      key="Plzen"
                      value="Plzen"
                    >
                      Plzen
                    </option>
                    <option
                      key="Ostrava"
                      value="Ostrava"
                    >
                      Ostrava
                    </option>
                    <option
                      key="Olomouc"
                      value="Olomouc"
                    >
                      Olomouc
                    </option>
                  </select>
                </div>
              </div>
            </div>
            <div
              className="mail-container container null"
            >
              <input
                className="form-input"
                id="mail"
                name="mail"
                onChange={[Function]}
                placeholder="mail@mail"
                type="mail"
                value=""
              />
            </div>
            <div
              className="message-container container"
            >
              <textarea
                className="form-input"
                cols="30"
                id="message"
                name="message"
                onChange={[Function]}
                placeholder="...message"
                rows="10"
                value=""
              />
            </div>
          </form>
        </Form>
        <Button
          class="contact-part-btn"
          fn={[Function]}
        >
          <button
            className="btn contact-part-btn "
            onClick={[Function]}
            type="button"
          >
            Contact us
          </button>
        </Button>
      </div>
    </div>
  </div>
</ContactPart>
`;

exports[`ContactPart component Check context Check without context 1`] = `
<ContactPart>
  <div
    className="contact-page stretched transition-all default"
    id="contacts"
  >
    <div
      className="bg"
    >
      <div
        className="contact-place"
      >
        <ContactTextIntro>
          <div
            className="contact-text-intro"
          >
            <h2>
              Contact Us
            </h2>
            <p>
              Lorem, ipsum dolor sit amet consectetur adipisicing elit. Maiores mollitia vel sapiente natus assumenda soluta earum explicabo, excepturi fugit minus.
            </p>
          </div>
        </ContactTextIntro>
        <Form
          categories={
            Array [
              "Fisrt",
              "Second",
              "Third",
            ]
          }
          category="First"
          cities={
            Array [
              "Praha",
              "Brno",
              "Plzen",
              "Ostrava",
              "Olomouc",
            ]
          }
          city="Praha"
          errObj={
            Object {
              "category": "",
              "city": "",
              "mail": "",
              "message": "",
              "name": "",
              "phone": "",
            }
          }
          mail=""
          message=""
          name=""
          phone="+420"
          setFormState={[Function]}
        >
          <form
            action=""
            className="contact-form"
            method=""
          >
            <div
              className="sides"
            >
              <div
                className="left-side"
              >
                <div
                  className="name-container container null"
                >
                  <input
                    className="form-input"
                    id="name"
                    name="name"
                    onChange={[Function]}
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="phone-container container null"
                >
                  <input
                    className="form-input"
                    id="phone-number"
                    name="phone"
                    onChange={[Function]}
                    type="tel"
                    value="+420"
                  />
                </div>
              </div>
              <div
                className="right-side"
              >
                <div
                  className="category-container container"
                >
                  <select
                    className="form-input"
                    id="category"
                    name="category"
                    onChange={[Function]}
                    value="First"
                  >
                    <option
                      key="Fisrt"
                      value="Fisrt"
                    >
                      Fisrt
                    </option>
                    <option
                      key="Second"
                      value="Second"
                    >
                      Second
                    </option>
                    <option
                      key="Third"
                      value="Third"
                    >
                      Third
                    </option>
                  </select>
                </div>
                <div
                  className="city-container container"
                >
                  <select
                    id="city"
                    name="city"
                    onChange={[Function]}
                    value="Praha"
                  >
                    <option
                      key="Praha"
                      value="Praha"
                    >
                      Praha
                    </option>
                    <option
                      key="Brno"
                      value="Brno"
                    >
                      Brno
                    </option>
                    <option
                      key="Plzen"
                      value="Plzen"
                    >
                      Plzen
                    </option>
                    <option
                      key="Ostrava"
                      value="Ostrava"
                    >
                      Ostrava
                    </option>
                    <option
                      key="Olomouc"
                      value="Olomouc"
                    >
                      Olomouc
                    </option>
                  </select>
                </div>
              </div>
            </div>
            <div
              className="mail-container container null"
            >
              <input
                className="form-input"
                id="mail"
                name="mail"
                onChange={[Function]}
                placeholder="mail@mail"
                type="mail"
                value=""
              />
            </div>
            <div
              className="message-container container"
            >
              <textarea
                className="form-input"
                cols="30"
                id="message"
                name="message"
                onChange={[Function]}
                placeholder="...message"
                rows="10"
                value=""
              />
            </div>
          </form>
        </Form>
        <Button
          class="contact-part-btn"
          fn={[Function]}
        >
          <button
            className="btn contact-part-btn "
            onClick={[Function]}
            type="button"
          >
            Contact us
          </button>
        </Button>
      </div>
    </div>
  </div>
</ContactPart>
`;
